Source/Initialization/WarpXInitData.cpp

```diff
@@ -1228,3 +1287,174 @@ void WarpX::CheckKnownIssues()
+
+    auto FC_chunk_data = FC.loadChunk<double>(chunk_offset,chunk_extent);
+    series.flush();
+    auto FC_data_host = FC_chunk_data.get();
+
+    // Load data to GPU
+    size_t total_extent = size_t(extent[0]) * extent[1] * extent[2];
+    amrex::Gpu::DeviceVector<double> FC_data_gpu(total_extent);
+    auto FC_data = FC_data_gpu.data();
+    amrex::Gpu::copy(amrex::Gpu::hostToDevice, FC_data_host, FC_data_host + total_extent, FC_data);
+
+    // Loop over boxes
+    for (MFIter mfi(*mf, TilingIfNotGPU()); mfi.isValid(); ++mfi)
+    {
+        amrex::Box box = mfi.growntilebox();
+        amrex::Box tb = mfi.tilebox(nodal_flag, mf->nGrowVect());
+        auto const& mffab = mf->array(mfi);
+
+        // Start ParallelFor
+        amrex::ParallelFor (tb,
+            [=] AMREX_GPU_DEVICE (int i, int j, int k) {
+                // i,j,k denote x,y,z indices in 3D xyz.
+                // i,j,k denote r,z,mode indices in 2D rz.
+
+                // ii is used for 2D RZ mode
+                int ii = i;
+#if defined(WARPX_DIM_RZ)
+                // In 2D RZ, i denoting r can be < 0
+                // but mirrored values should be assigned.
+                // Namely, mffab(i) = FC_data[-i] when i<0.
+                if (i<0) {ii = -i;}
+#endif
+
+                // Physical coordinates of the grid point
+                // 0,1,2 denote x,y,z in 3D xyz.
+                // 0,1 denote r,z in 2D rz.
+                amrex::Real x0, x1;
+                if ( box.type(0)==amrex::IndexType::CellIndex::NODE )
+                     { x0 = real_box.lo(0) + ii*dx[0]; }
+                else { x0 = real_box.lo(0) + ii*dx[0] + 0.5*dx[0]; }
+                if ( box.type(1)==amrex::IndexType::CellIndex::NODE )
+                     { x1 = real_box.lo(1) + j*dx[1]; }
+                else { x1 = real_box.lo(1) + j*dx[1] + 0.5*dx[1]; }
+
+                // Get index of the external field array
+                int const ix0 = floor( (x0-offset0)/d0 );
+                int const ix1 = floor( (x1-offset1)/d1 );
+
+                // Get coordinates of external grid point
+                amrex::Real const xx0 = offset0 + ix0*d0;
+                amrex::Real const xx1 = offset1 + ix1*d1;
+
+#if defined(WARPX_DIM_3D)
+                amrex::Real x2;
+                if ( box.type(2)==amrex::IndexType::CellIndex::NODE )
+                     { x2 = real_box.lo(2) + k*dx[2]; }
+                else { x2 = real_box.lo(2) + k*dx[2] + 0.5*dx[2]; }
+                int const ix2 = floor( (x2-offset2)/d2 );
+                amrex::Real const xx2 = offset2 + ix2*d2;
+#endif
+
+#if defined(WARPX_DIM_XZ) || defined(WARPX_DIM_RZ)
+                amrex::Array4<double> fc_array(FC_data, {0,0,0}, {extent0, extent2, extent1}, 1);
+                double
+                    f00 = fc_array(0, ix1  , ix0  ),
+                    f01 = fc_array(0, ix1  , ix0+1),
+                    f10 = fc_array(0, ix1+1, ix0  ),
+                    f11 = fc_array(0, ix1+1, ix0+1);
```

Comment:
Here I just want to double check.

I see that the order of the indices to access `fc_array` in the 2D/RZ cases is:
- 1st index always 0
- 2nd index related to `ix1`
- 3rd index related to `ix0`

and the extents are passed in the order `{extent0, extent2, extent1}`.

On the other hand, the order of the indices to access `fc_array` in the 3D cases is:
- 1st index related to `ix2`
- 2nd index related to `ix1`
- 3rd index related to `ix0`

and the extents are passed in the order `{extent2, extent1, extent0}`.

Somehow the order of the indices vs the order of the extents seem to me to match in 3D but not in 2D/RZ.
Could you help me clarify why this would be the case?

Reply:
In the RZ case, 1st index is theta, which is currently asserted as unused because it's not implemented here.

Reply:
For complete transparency, this code is utterly untested in the XZ case, because that seemingly garnered no interest from anyone who's been paid to look into this functionality.

Reply:
I could add an assertion against that, too, I guess, and leave it for someone who wants to use it to do the testing.

Reply:
Thanks for the comments. I discussed this briefly with @RemiLehe offline, we think it could be nice to choose variable names that create a bit less confusion, with respect to these index "tags" such as `0`, `1` and `2` added to variable names. Maybe we could add `r`, `z` and `mode` suffixes instead and use them only for RZ with `#ifdef` blocks. Or other solutions would be welcome, too. This might require a bit of live brainstorming with you and @RemiLehe.